qiskit_experiments/characterization/qubit_spectroscopy.py

```diff
@@ -369,14 +366,41 @@ def __init__(
         if len(frequencies) < 3:
             raise QiskitError("Spectroscopy requires at least three frequencies.")
 
-        if unit not in self.__units__:
-            raise QiskitError(f"Unsupported unit: {unit}.")
+        if unit == "Hz":
+            self._frequencies = frequencies
+        else:
+            self._frequencies = [apply_prefix(freq, unit) for freq in frequencies]
 
-        self._frequencies = [freq * self.__units__[unit] for freq in frequencies]
         self._absolute = absolute
 
         super().__init__([qubit])
 
+    def _schedule(self, backend: Optional[Backend] = None) -> Tuple[pulse.ScheduleBlock, Parameter]:
```

Comment:
```suggestion
    def _spect_gate_schedule(self, backend: Optional[Backend] = None) -> Tuple[pulse.ScheduleBlock, Parameter]:
```
`_schedule` might be confusing. Because this can indicate the schedule of transpiled circuit.